zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -153,10 +168,14 @@ impl<F: Field> ConstraintBuilder<F> {
         // in the current row differs from the previous row.
         self.condition(q.first_access(), |cb| {
             cb.require_zero(
-                "first access reads don't change value",
-                q.is_read() * (q.rw_table.value.clone() - q.initial_value()),
+                "first access reads don't change value (hi)",
+                q.is_read() * (q.rw_table.value.hi() - q.initial_value().hi()),
+            );
+            cb.require_zero(
+                "first access reads don't change value (lo)",
+                q.is_read() * (q.rw_table.value.lo() - q.initial_value().lo()),
```

Comment:
line 171, 174 (and few ditto below) can be combined into
```
cb.require_zero_word(
    "first access reads don't change value",
    q.rw_table.value.sub_unchecked(q.initial_value()).mul_selector(q.is_read())
)
```

Reply:
> Why not implement
> 
> ```
> impl<F: Field> std::ops::Add for Word<Expression<F>>{
>     type Output = Word<Expression<F>>;
>     
>     fn add(self, rhs: Word<Expression<F>>) -> Self::Output {
>         Word::new([self.lo() + rhs.lo(), self.hi() + rhs.hi()]) 
>     }
> } 
> 
> impl<F: Field> std::ops::Sub for Word<Expression<F>>{
>     type Output = Word<Expression<F>>;
>     
>     fn sub(self, rhs: Word<Expression<F>>) -> Self::Output {
>         Word::new([self.lo() - rhs.lo(), self.hi() - rhs.hi()])
>     }
> } 
> ```
> 
> ?

Because by spec lo/hi should within 128bit, and purely add/sub we can't avoid overflow/underflow issue